@xen-orchestra/xapi/host.mjs

```diff
@@ -162,6 +162,33 @@ class Host {
 
     return ipmiSensorsByDataType
   }
+
+  async getBiosInfo(ref, { cache }) {
+    const biosData = await this.call('host.get_bios_strings', ref)
+    const { 'bios-version': currentBiosVersion, 'system-product-name': hostServerName } = biosData
+
+    if (!cache.has('servers')) {
+      const response = await fetch(
+        'https://pictures.2cr.si/Images_site_web_Odoo/Pages_produit/VATES-BIOS_BMC_last-version.json'
+      )
+      const json = await response.json()
+      const servers = json[0]['2CRSi_Servers']
+      cache.set('servers', servers)
+    }
+
+    const parsedData = cache.get('servers')
+
+    const serverData = parsedData.find(server => server.Server_Name === hostServerName)
+
+    if (serverData === undefined) {
+      return
+    }
+
+    const { 'BIOS-Version': latestBiosVersion, 'BIOS-link': biosLink } = serverData
+    const isUpToDate = currentBiosVersion === latestBiosVersion
+
+    return { currentBiosVersion, latestBiosVersion, biosLink, isUpToDate }
+  }
```

Comment:
IMHO, this code is pretty specific to 2CRSI, if you think it could be generalize, keep it there, otherwise I suggest we move into `xo-server`.

Let's try to keep this lib as clean as we can.